logging/logmanager/plugin_impl_log_manager.go

```diff
@@ -91,22 +93,40 @@ func (lm *Plugin) Init() error {
 		}
 		lm.Log.Debugf("logs config: %+v", lm.Conf)
 
+		// Handle default log level
 		if lm.Conf.DefaultLevel != "" {
 			if err := lm.LogRegistry.SetLevel("default", lm.Conf.DefaultLevel); err != nil {
-				lm.Log.Warn("setting default log level failed:", err)
+				lm.Log.Warnf("setting default log level failed: %v", err)
 			} else {
-				lm.Log.Debugf("default log level to %q", lm.Conf.DefaultLevel)
+				// All loggers created up to this point were created with initial log level set (defined
+				// via INITIAL_LOGLVL env. variable with value 'info' by default), so at first, let's set default
+				// log level for all of them.
+				for loggerName := range lm.LogRegistry.ListLoggers() {
+					logger, exists := lm.LogRegistry.Lookup(loggerName)
+					if !exists {
+						continue
+					}
+					logger.SetLevel(stringToLogLevel(lm.Conf.DefaultLevel))
+				}
 			}
 		}
 
-		// try to set log levels (note, not all of them might exist yet)
-		for _, cfgLogger := range lm.Conf.Loggers {
-			if err := lm.LogRegistry.SetLevel(cfgLogger.Name, cfgLogger.Level); err != nil {
-				//intentionally just log warn & not propagate the error (it is minor thing to interrupt startup)
-				lm.Log.Warn("setting level failed:", err)
+		// Handle config file log levels
+		for _, logCfgEntry := range lm.Conf.Loggers {
+			// Put log/level entries from configuration file to the registry.
+			if err := lm.LogRegistry.SetLevel(logCfgEntry.Name, logCfgEntry.Level); err != nil {
+				// Intentionally just log warn & not propagate the error (it is minor thing to interrupt startup)
+				lm.Log.Warnf("setting log level %s for logger %s failed: %v", logCfgEntry.Level,
+					logCfgEntry.Name, err)
+			} else {
+				// Check whether such a logger does not exist. If so, set the level now (override default value).
+				logger, exists := lm.LogRegistry.Lookup(logCfgEntry.Name)
+				if !exists {
+					continue
+				}
+				logger.SetLevel(stringToLogLevel(logCfgEntry.Level))
```

Comment:
This else clause is not needed. You are already setting the level via `LogRegistry.SetLevel` which does the lookup and `Logger.SetLevel` inside.